exit/test_exit.c

```diff
@@ -0,0 +1,811 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    POSIX.1-2017 standard library functions tests
+ *    HEADER:
+ *    - stdlib.h
+ *    - unistd.h
+ *    TESTED:
+ *    - exit()
+ *    - _exit() (_Exit equivalent)
+ *    - atexit()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <pthread.h>
+
+#include <unity_fixture.h>
+
+#define TEST_EXIT_PATH      "/tmp/exit_test_file"
+#define TEST_EXIT_STR       "test123"
+#define TEST_EXIT_DUMMY_VAL 12345678
+
+/*
+	Aspects required by POSIX, which wasn't tested:
+		- all open named semaphores in the calling process shall be closed
+		  -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/806
+
+		- threads terminated by a call to _Exit() or _exit() shall not invoke their cancellation cleanup handlers
+		  -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/827
+
+		- if the exit of the process causes a process group to become orphaned, and if any member of the newly-orphaned
+		  process group is stopped, then a SIGHUP signal followed by a SIGCONT signal shall be sent to each process in
+		  the newly-orphaned process group -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/809
+
+		- If the process is a controlling process, the SIGHUP signal shall be sent to each process in the foreground
+		  process group of the controlling terminal belonging to the calling process.
+		  If the process is a controlling process, the controlling terminal associated with the session shall be
+		  disassociated from the session, allowing it to be acquired by a new controlling process
+		  -> ???
+
+		- The full value of status shall be available from waitid() and in the siginfo_t passed to a signal handler
+		  for SIGCHLD -> https://github.com/phoenix-rtos/phoenix-rtos-project/issues/844,
+		  https://github.com/phoenix-rtos/phoenix-rtos-project/issues/845
+ */
+
+
```

Comment:
What about memory mapping?
![Screenshot from 2023-09-29 11-34-27](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/2e976334-0cdf-4b83-ae43-4b3693464d49)
Please research it and leave a comment/issue about it.